lib/backend/k8s/resources/profile.go

```diff
@@ -109,24 +171,54 @@ func (c *profileClient) List(ctx context.Context, list model.ListInterface, revi
 		}, nil
 	}
 
+	nsRev, saRev, err := c.splitRev(revision)
+	if err != nil {
+		return nil, err
+	}
+
 	// Otherwise, enumerate all.
-	namespaces, err := c.clientSet.CoreV1().Namespaces().List(metav1.ListOptions{ResourceVersion: revision})
+	namespaces, err := c.clientSet.CoreV1().Namespaces().List(metav1.ListOptions{ResourceVersion: nsRev})
 	if err != nil {
 		return nil, K8sErrorToCalico(err, nl)
 	}
 
 	// For each Namespace, return a profile.
 	for _, ns := range namespaces.Items {
-		kvp, err := c.converter.NamespaceToProfile(&ns)
+		kvp, err := c.getNsKv(&ns)
 		if err != nil {
 			log.Errorf("Unable to convert k8s Namespace to Calico Profile: Namespace=%s: %v", ns.Name, err)
 			continue
 		}
 		kvps = append(kvps, kvp)
 	}
+
+	if c.AlphaSA == false {
+		return &model.KVPairList{
+			KVPairs:  kvps,
+			Revision: namespaces.ResourceVersion,
```

Comment:
Is this correct?  I think everywhere else you assume the revision will be a "dual revision" even if the feature is not enabled.